rye/src/cli/add.rs

```diff
@@ -468,23 +468,29 @@ fn resolve_requirements_with_uv(
         if let Ok(dt) = env::var("__RYE_UV_EXCLUDE_NEWER") {
             cmd.arg("--exclude-newer").arg(dt);
         }
-        let sources = ExpandedSources::from_sources(&pyproject_toml.sources()?)?;
-        sources.add_as_pip_args(&mut cmd);
-        let mut child = cmd
-            .stdin(Stdio::piped())
-            .stdout(Stdio::piped())
-            .stderr(Stdio::piped())
-            .spawn()?;
-        let child_stdin = child.stdin.as_mut().unwrap();
-        writeln!(child_stdin, "{}", req)?;
-
-        let rv = child.wait_with_output()?;
-        if !rv.status.success() {
-            let log = String::from_utf8_lossy(&rv.stderr);
-            bail!("failed to resolve packages:\n{}", log);
-        }
 
-        let mut new_req: Requirement = String::from_utf8_lossy(&rv.stdout).parse()?;
+        let mut new_req = match req.version_or_url {
+            Some(VersionOrUrl::Url(_)) => req.clone(),
```

Comment:
I wanted to look into why the requirement would need to make a trip through `uv` to begin with. I'm sure I'm forgetting something obvious, but this at least demonstrates where the issue might be coming from.

Reply:
See [comment](https://github.com/astral-sh/rye/pull/817#issuecomment-1975280848)

Reply:
As the code is here, you skip using the result from uv for URLs and that makes sense (If I understand the code correctly) - but should we then also skip using uv at all? I don't know if it's important for the add command do some kind of validation of the arguments like this (checking that they exist)

Reply:
Yup we seem to be on the same page. Initially it seems like we can just skip it since these are direct references, but I want to understand the big picture a little better.

Thanks for looking at this!